libdrgn/dwarf_index.c

```diff
@@ -2346,10 +2444,14 @@ static struct drgn_error *index_cu_second_pass(struct drgn_dwarf_index *dindex,
 				file_name_hash = cu->file_name_hashes[decl_file - 1];
 			else
 				file_name_hash = 0;
-			if ((err = index_die(dindex, name,
-					     insn & DIE_FLAG_TAG_MASK,
+
+			struct drgn_dwarf_index_namespace *new_ns = NULL;
+			if ((err = index_die(ns, name,
+					     tag,
 					     file_name_hash, module,
-					     die_offset)))
+					     die_offset,
+					     die_offset + debug_info_buffer,
```

Comment:
This is a little inelegant, I could just store the offset instead of the ptr, but then I would need to have logic to detect if it's being passed in again (debug_info_buffer isn't available outside of this fn).

Reply:
Hm, if we store the offset in the `pending_cu`, can't we get the pointer as `section_ptr(pending_cu->cu->userdata->debug_info, 0) + pending_cu->offset`?

Reply:
I guess that would work, I was worried it would just be a little messy at that point then, I'll switch it over.